extensions/rk-panthor.sh

```diff
@@ -1 +1,26 @@
-declare -g DEFAULT_OVERLAYS=panthor-gpu
+#
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (c) 2024 Ricardo Pardini <ricardo@pardini.net>
+# This file is a part of the Armbian Build Framework https://github.com/armbian/build/
+#
+
+# Enables panthor-gpu overlay and oibaf-mesa extension. Only meant for legacy/vendor branches of Rockchip boards.
+enable_extension "mesa-oibaf" # Enable OIBAF repo for mainline mesa
+
+function extension_prepare_config__rk_panthor() {
+	display_alert "Preparing rk-panthor" "${EXTENSION}" "info"
+	EXTRA_IMAGE_SUFFIXES+=("-panthor") # Add to the image suffix. # global array
+
+	# Enable panthor overlay by default
+	declare -g DEFAULT_OVERLAYS="panthor-gpu"
+
+	[[ "${BUILDING_IMAGE}" != "yes" ]] && return 0
+
+	if [[ "${LINUXFAMILY}" != "rockchip-rk3588" && "${LINUXFAMILY}" != "rk35xx" ]]; then
```

Comment:
panthor-gpu overlay can be only enabled in `rk35xx-vendor`. When `LINUXFAMILY` is `rockchip-rk3588`, it is mainline edge branch or collabora. So it's better to limit this extension to `LINUXFAMILY` defined as `rk35xx` and branch `vendor`.

Reply:
Indeed; the problem here is that we have both `rk35xx` _and_ `rockchip-rk3588` with `vendor` branch. (old problem)

Either way, no `current` or `edge` branch is allowed in the check after this one, so it should be safe.